example/fundamentals/tasks/6-workers/build.mill

```diff
@@ -30,20 +36,25 @@ class CompressWorker(dest: os.Path) {
   val cache = collection.mutable.Map.empty[Int, Array[Byte]]
   def compress(name: String, bytes: Array[Byte]): Array[Byte] = {
     val hash = Arrays.hashCode(bytes)
-    if (!cache.contains(hash)) {
+    if (!synchronized(cache.contains(hash))) {
```

Comment:
Naive 'user level' complaint:

I'm not a fan the 'synchronized' in an example. Its so easy to get wrong and I'm always having a hard time to convince myself that it is correct.

What I would love instead is having the concurrent map operations on it:
`cache.computeIfAbsent(key, (key)=>createTheExpensiveThing())`

Note again: This is from a doc/end user perspective and not the internal use.

Reply:
Oh...this is a negative example / problem statement =). So,yes, then it makes sense.